ui/app/src/components/shared/ConfirmUpgradeResource.tsx

```diff
@@ -0,0 +1,113 @@
+import { Dialog, DialogFooter, PrimaryButton, DialogType, Spinner, Dropdown, MessageBar, MessageBarType, DropdownMenuItemType, IDropdownOption, Icon, Stack, Label, IconButton, IDropdownProps } from '@fluentui/react';
+import React, { useContext, useState } from 'react';
+import { AvailableUpgrade, Resource } from '../../models/resource';
+import { HttpMethod, ResultType, useAuthApiCall } from '../../hooks/useAuthApiCall';
+import { WorkspaceContext } from '../../contexts/WorkspaceContext';
+import { ResourceType } from '../../models/resourceType';
+import { APIError } from '../../models/exceptions';
+import { LoadingState } from '../../models/loadingState';
+import { ExceptionLayout } from './ExceptionLayout';
+import { useAppDispatch } from '../../hooks/customReduxHooks';
+import { addUpdateOperation } from '../shared/notifications/operationsSlice';
+
+interface ConfirmUpgradeProps {
+  resource: Resource,
+  onDismiss: () => void
+}
+
+export const ConfirmUpgradeResource: React.FunctionComponent<ConfirmUpgradeProps> = (props: ConfirmUpgradeProps) => {
+  const apiCall = useAuthApiCall();
+  const [selectedVersion, setSelectedVersion] = useState("")
+  const [apiError, setApiError] = useState({} as APIError);
+  const [loading, setLoading] = useState(LoadingState.Ok);
+  const workspaceCtx = useContext(WorkspaceContext);
+  const dispatch = useAppDispatch();
+
+  const upgradeProps = {
+    type: DialogType.normal,
+    title: `Upgrade Template Version?`,
+    closeButtonAriaLabel: 'Close',
+    subText: `Are you sure you want upgrade the template version of ${props.resource.properties.display_name} from version ${props.resource.templateVersion}?`,
+  };
+
+  const dialogStyles = { main: { maxWidth: 450 } };
+  const modalProps = {
+    titleAriaId: 'labelId',
+    subtitleAriaId: 'subTextId',
+    isBlocking: true,
+    styles: dialogStyles
+  };
+
+  const wsAuth = (props.resource.resourceType === ResourceType.WorkspaceService || props.resource.resourceType === ResourceType.UserResource);
+
+  const upgradeCall = async () => {
+    setLoading(LoadingState.Loading);
```

Comment:
`nit` we tend to use "loading" and "loadingState" for indicating whether a page/component has loaded after navigation. For API request statuses, maybe call is `requestStatus` or something more specific to a request

Reply:
👍  renamed to `requestLoadingState`